aptos-node/src/lib.rs

```diff
@@ -889,9 +889,20 @@ pub fn setup_environment(
     })
 }
 
+pub const ERROR_MSG_BAD_FEATURE_FLAGS: &str = r#"
+aptos-node was compiled with feature flags that shouldn't be enabled.
+
+This is caused by cargo's feature unification.
+When you compile two crates with a shared dependency, if one enables a feature flag for the dependency, then it is also enabled for the other crate.
+
+PLEASE RECOMPILE APTOS-NODE SEPARATELY using the following command:
+    cargo build --package aptos-node
```

Comment:
nit: do we need to call out cargo clean?

Reply:
Doesn't seem necessary. I tried running `cargo build --package aptos-node` right after running `cargo build --bin` (at workspace root) and the binary worked.